groovy/postInit/chemistry/organic_chemistry/BiofuelChain.groovy

```diff
@@ -0,0 +1,310 @@
+            .duration(20)
+            .EUt(30)
+            .buildAndRegister()
+
+        // Neutralization of Crude FAXEs
+        MIXER.recipeBuilder()
+            .fluidInputs(fluid('diluted_hydrochloric_acid') * 100)
+            .fluidInputs(alcohol.getCrudeEster(2100))
+            .fluidOutputs(alcohol.getNeutralizedEster(2100))
+            .fluidOutputs(fluid('diluted_saltwater') * 100)
+            .duration(15)
+            .EUt(7)
+            .buildAndRegister()
+
+        // Distillation of Neutralized FAXEs
+        DT.recipeBuilder()
+            .fluidInputs(alcohol.getNeutralizedEster(3500))
+            .fluidOutputs(fluid('unscrubbed_bio_diesel') * 3000)
+            .fluidOutputs(alcohol.get(500))
+            .duration(tpbb * 3)
+            .EUt(30)
+            .buildAndRegister()
+
+        // Washing of Bio Diesel with heated water
+        MIXER.recipeBuilder()
+            .fluidInputs(fluid('gtfo_heated_water') * 1000)
+            .fluidInputs(fluid('unscrubbed_bio_diesel') * 3000)
+            .fluidOutputs(fluid('wastewater') * 1000)
+            .fluidOutputs(fluid('moist_bio_diesel') * 3000)
+            .duration(tpbb * 3)
+            .EUt(7)
+            .buildAndRegister()
+
+        // Drying
+        DRYER.recipeBuilder()
+            .fluidInputs(fluid('moist_bio_diesel') * 1000)
+            .fluidOutputs(fluid('bio_diesel') * 1000)
+            .duration(tpbb)
+            .EUt(30)
+            .buildAndRegister()
+}
+
+// tick per 1/4 bucket of n-paraffin
+int tpbnp = 20
+
+// -------- Green Diesel Chain --------
+// Pretreatment to remove undesired pigments
+// such as carotenoids and chlorophyll via decolorization
+grade1Lipids.forEach { lipid ->
+        MIXER.recipeBuilder()
+            .inputs(ore('dustAluminiumSilicate'))
+            .notConsumable(fluid('phosphoric_acid') * 500)
+            .fluidInputs(lipid.get())
+            .fluidOutputs(fluid('bleaching_lipid_mix') * 1000)
+            .duration(120)
+            .EUt(30)
+            .buildAndRegister()
+
+        // Separation of Al2SiO5
+        SIFTER.recipeBuilder()
+            .fluidInputs(fluid('bleaching_lipid_mix') * 1000)
+            .outputs(metaitem('dustAluminiumSilicate'))
+            .fluidOutputs(fluid('bleached_lipid') * 1000)
+            .duration(20)
+            .EUt(30)
+            .buildAndRegister()
+}
+
+// Preheat the bleached lipid for the hydrotreating process
+FLUID_HEATER.recipeBuilder()
+        .fluidInputs(fluid('bleached_lipid') * 1000)
+        .fluidOutputs(fluid('heated_lipid') * 1000)
+        .duration(30)
+        .EUt(30)
+        .buildAndRegister()
+
+// Hydrotreating heated lipid
+TBR.recipeBuilder()
+        .fluidInputs(fluid('heated_lipid') * 250)
+        .fluidInputs(fluid('hydrogen') * 8000)
+        .notConsumable(ore('dustHydrotreatingCatalyst')) // Since molybdenum is EV
+        .fluidOutputs(fluid('hydrodeoxygenation_reaction_outlet') * 7000)
```

Comment:
this may be a bit cumbersome of a name, consider hydrotreated lipid

Reply:
k